Bulldozer.CSV/Maps/Metrics.cs

```diff
@@ -334,5 +335,615 @@ private void SaveMetrics( List<MetricValue> metricValues )
+                            if ( entity == null )
+                            {
+                                errors += $"{DateTime.Now}, MetricValuePartition, EntityId {metricValuePartitionImport.EntityId} not found for EntityTypeName {entityType.Name}. MetricPartitionValue for {metricValuePartitionImport.CsvMetricValueId} metric value was skipped.\r\n";
+                                continue;
+                            }
+                            var newMetricValuePartition = new MetricValuePartition
+                            {
+                                MetricValueId = metricValuePartitionImport.MetricValue.Id,
+                                MetricPartitionId = metricValuePartitionImport.MetricPartition.Id,
+                                EntityId = entity.Id,
+                                ForeignKey = metricValuePartitionImport.ForeignKey
+                            };
+
+                            metricValuePartionsToInsert.Add( newMetricValuePartition );
+                        }
+
+                    }
+                }
+
+                if ( metricValuePartionsToInsert.Count > 0 )
+                {
+                    ReportProgress( 0, $"Begin processing { entityType.FriendlyName } type Metric Value Partitions." );
+                }
+
+                // Slice Metric Value Partition data into chunks and process
+                var workingMetricValuePartitionImportList = metricValuePartionsToInsert;
+                var metricValuePartitionsRemainingToProcess = workingMetricValuePartitionImportList.Count();
+                var completedMetricValuePartitions = 0;
+
+                while ( metricValuePartitionsRemainingToProcess > 0 )
+                {
+                    if ( completedMetricValuePartitions > 0 && completedMetricValuePartitions % ( this.DefaultChunkSize * 10 ) < 1 )
+                    {
+                        ReportProgress( 0, $"{completedMetricValuePartitions} Metric Values processed." );
+                    }
+
+                    if ( completedMetricValuePartitions % this.DefaultChunkSize < 1 )
+                    {
+                        var csvChunk = workingMetricValuePartitionImportList.Take( Math.Min( this.DefaultChunkSize, workingMetricValuePartitionImportList.Count ) ).ToList();
+                        rockContext.BulkInsert( csvChunk );
+                        completedMetricValuePartitions += csvChunk.Count;
+                        metricValuePartitionsRemainingToProcess -= csvChunk.Count;
+                        workingMetricValuePartitionImportList.RemoveRange( 0, csvChunk.Count );
+                        ReportPartialProgress();
+                    }
+                }
+            }
+            if ( errors.IsNotNullOrWhiteSpace() )
+            {
+                LogException( null, errors, hasMultipleErrors: true );
+            }
+
+            return completedMetricValues;
+        }
```

Comment:
Ensure this is at the right level of this method.

Reply:
confirmed